NAudio.Midi/Midi/MidiIn.cs

```diff
@@ -145,6 +193,19 @@ protected virtual void Dispose(bool disposing)
             if(!this.disposed) 
             {
                 //if(disposing) Components.Dispose();
+
+                //  Reset in order to release any Sysex buffers
+                //  We can't Unprepare and free them until they are flushed out. Neither can we close the handle.
+                MmException.Try(MidiInterop.midiInReset(hMidiIn), "midiInReset");
+
+                //  Free up all created and allocated buffers for incoming Sysex messages
+                foreach (var lpHeader in SysexBufferHeaders)
+                {
+                    MidiInterop.MIDIHDR hdr = (MidiInterop.MIDIHDR)Marshal.PtrToStructure(lpHeader, typeof(MidiInterop.MIDIHDR));
+                    MmException.Try(MidiInterop.midiInUnprepareHeader(hMidiIn, lpHeader, Marshal.SizeOf(typeof(MidiInterop.MIDIHDR))), "midiInPrepareHeader");
+                    Marshal.FreeHGlobal(hdr.lpData);
+                    Marshal.FreeHGlobal(lpHeader);
+                }
                 MidiInterop.midiInClose(hMidiIn);
```

Comment:
also perhaps here set SysexBufferHeaders to a 0 length array /null here

Reply:
Done - as added defensive protection against double disposal